python/mxnet/gluon/block.py

```diff
@@ -967,6 +968,27 @@ def _build_cache(self, *args):
                 self._cached_op_args.append((False, params[name]))
         flags = [('data_indices', data_indices), ('param_indices', param_indices)] + \
                 self._flags
+
+        args, _ = _flatten(args, "input")
+        args_without_none = [ele for ele in args if ele is not None]
+        try:
+            for is_arg, i in self._cached_op_args:
+                if not is_arg:
+                    tmp = i.data()
```

Comment:
can we remove assignment and just call `i.data()`

Reply:
Done.